### PR TITLE
samples: hci_rpmsg: Use release configuration by default

### PR DESCRIPTION
**Pull request by MarekPieta**
Change removes logs and assertions from default configuration. They can be optionally enabled with debug_overlay.conf.

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: samples/bluetooth/hci_rpmsg/prj.conf [hunk 1]
```diff
@@ -10,9 +8,4 @@ CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=512
 CONFIG_BT=y
 CONFIG_BT_HCI_RAW=y
 CONFIG_BT_MAX_CONN=16
-CONFIG_BT_CTLR_ASSERT_HANDLER=y
```
Comment (cvinayak) [inline samples/bluetooth/hci_rpmsg/prj.conf:13]: This is required to generate HCI Hardware Error Event.

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Review by KAGA164 — APPROVED** [review]

File: samples/bluetooth/hci_rpmsg/prj.conf (hunk 1 above)
Comment (MarekPieta) [reply samples/bluetooth/hci_rpmsg/prj.conf:13]: In hci_rpmsg sample, the application-defined handler actually performs BT_ASSERT_MSG. In the default configuration logger is disabled, so I think it actually leads to the same behavior as for "default" handling mechanism. Did I overlook something ?
I noticed dedicated event in hci_uart sample.

Comment (MarekPieta) [reply samples/bluetooth/hci_rpmsg/prj.conf:13]: Hi @cvinayak, could you look at this one again ?

**Comment by pdunaj** [comment]
Hi @cvinayak , can you have a look?

File: samples/bluetooth/hci_rpmsg/prj.conf (hunk 1 above)
Comment (cvinayak) [reply samples/bluetooth/hci_rpmsg/prj.conf:13]: As the title of this PR wants to have release configuration, the `CONFIG_BT_CTLR_ASSERT_HANDLER=y`  should not be removed. Instead, the assert should be encoded as "HCI Hardware Error event" and sent to application core. hci_uart sample on application core plus the hci_rpmsg sample on network core will be used for conformance testing and it is needed to capture asserts, if any, in the HCI trace in the conformance tester.

Adding the HCI Hardware Error event is outside the scope of this PR, but `CONFIG_BT_CTLR_ASSERT_HANDLER` should not be removed, as `CONFIG_BT_CTLR_ASSERT_HANDLER` is a release configuration for hci_rpmsg sample.

**Comment by MarekPieta** [comment]
I noticed that somehow turning off logs on network core causes HCI command timeouts during bonding (I observed BT assertion fail on bt_hci_cmd_send_sync). The problem replicates always when bonding with my Android device. I would like to fix it before merging this one.

**Comment by MarekPieta** [comment]
On another sample the problem also replicates with logging enabled on network core. Somehow it disappears when I enable assertions. It looks like some sort of strange race condition on network core. Sometimes only enabling assertions helps too.

**Review by KAGA164 — CHANGES_REQUESTED** [review]
I did a check on it. It looks like after disabling logging the data form rpmsq transport is processed a little faster than with logs enabled so I discovered that we drop some HCI command due to lack of a free command buffer. The simplest solution to fix this is increasing a count a available buffers by `CONFIG_BT_BUF_CMD_TX_COUNT` option

File: samples/bluetooth/hci_rpmsg/prj.conf (hunk 1 above)
Comment (MarekPieta) [reply samples/bluetooth/hci_rpmsg/prj.conf:13]: Added `CONFIG_BT_CTLR_ASSERT_HANDLER` back to default configuration. Thank you for the explanation.

**Comment by MarekPieta** [comment]
Thanks for investigation. @KAGA164. The suggestion was applied to default configuration of the sample.

**Review by cvinayak — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]

**Pull request merged**